sale_fixed_discount/models/sale_order.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2017-20 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import json
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends(
+        "order_line.tax_id", "order_line.price_unit", "amount_total", "amount_untaxed"
+    )
+    def _compute_tax_totals_json(self):
+        def compute_taxes(order_line):
+            real_price = order_line.price_unit * (
+                1 - (order_line.discount or 0.0) / 100.0
+            ) - (order_line.discount_fixed or 0.0)
+            twicked_price = real_price / (1 - (order_line.discount or 0.0) / 100.0)
+            price = (
+                order_line.price_unit * (1 - (order_line.discount or 0.0) / 100.0)
+                if not order_line.discount_fixed
+                else twicked_price
+            )
```

Comment:
Does anyone understand this code?

I would say this line would be fine:
price = order_line.price_unit * (1 - (order_line.discount or 0.0) / 100.0) - (order_line.discount_fixed or 0.0)

Reply:
@pedrobaeza do you have an opinion on this? 
Using the UI you can only insert discount or discount_fixed. So my simplification should be sufficient.

Via the API discount and discount_fixed might be possible and the current implementation of twicked_price is like this:
twicked_price = (100€ * 0,9 - 20€) / 0.9 = 77,78€ 
While my suggested calculation would be:
price = 100€ * 0,9 - 20€ = 70€ which would be an addition of the discounts. 

I also have my concerns that the tax is calculated correctly when discount and discount_fixed is provided.
https://github.com/OCA/account-invoicing/issues/1107

Reply:
Add then a constraint (SQL or Python) for not being able to save both fields with value.

Reply:
Now you are saying it, there is already such a constraint.

Reply:
Then no problem, isn't it? :)

Reply:
@pedrobaeza everything is done from my side. Would you like to review / merge this PR?